src/MUSI6106Exec/MUSI6106Exec.cpp

```diff
@@ -16,29 +15,152 @@ void    showClInfo ();
+        return -1;
+    }
+    else
+    {
+        sInputFilePath = argv[1];
+        sOutputFilePath = argv[2];
+        fDelayInSec = atof(argv[3]);
+        fModWidthInSec = atof(argv[4]);
+        fFreqModInHz = atof(argv[5]);
+    }
+
+
+
+    //============================================================================
+    // open the input wave file
+    //============================================================================
+
+    CAudioFileIf::create(phInputAudioFile);
+    phInputAudioFile->openFile(sInputFilePath,
+                               CAudioFileIf::kFileRead);
+
+    if (!phInputAudioFile->isOpen())
+    {
+        cout << "Input wave file open error!" << endl;
+        CAudioFileIf::destroy(phInputAudioFile);
+    }
+    phInputAudioFile->getFileSpec(stFileSpec);
+
+
+    //============================================================================
+    // Open the output audio file
+    //============================================================================
+
+    CAudioFileIf::create(phOutputAudioFile);
+    phOutputAudioFile -> openFile(sOutputFilePath,
+                                  CAudioFileIf::kFileWrite,
+                                  &stFileSpec);
+
+    if (!phOutputAudioFile->isOpen())
+    {
+        cout << "Output wave file open error!" << endl;
+        CAudioFileIf::destroy(phOutputAudioFile);
+    }
+
+    // allocate memory
+    ppfInputAudioData = new float*[stFileSpec.iNumChannels];
+    ppfOutputAudioData = new float* [stFileSpec.iNumChannels];
+    for (int i = 0; i < stFileSpec.iNumChannels; i++)
+    {
+        ppfInputAudioData[i] = new float[kBlockSize];
+        ppfOutputAudioData[i] = new float[kBlockSize];
+    }
+
+    // Destroy and clear memory
+    if (ppfInputAudioData[0] == nullptr || ppfOutputAudioData[0] == nullptr)
+    {
+        CAudioFileIf::destroy(phInputAudioFile);
+        CAudioFileIf::destroy(phOutputAudioFile);
+    }
+
+    //============================================================================
+    // Create Vibrato and allocate memory
+    //============================================================================
+
+    CVibratoIf::create(pVibrato);
+    pVibrato -> init(fDelayInSec, fFreqModInHz,
+                     stFileSpec.iNumChannels,
+                     stFileSpec.fSampleRateInHz,
+                     fModWidthInSec);
 
-    pCRingBuff = new CRingBuffer<float>(kBlockSize);
 
-    for (int i = 0; i < 5; i++)
+    //============================================================================
+    // Read, Filter and Write Audio
+    //============================================================================
+
+
+    while (!phInputAudioFile->isEof())
     {
-        pCRingBuff->putPostInc(1.F*i);
+        // set block length variable
+        long long int iNumFrames = kBlockSize;
+
+        // read data (iNumOfFrames might be updated!)
+        phInputAudioFile->readData(ppfInputAudioData, iNumFrames);
```

Comment:
Since `iNumFrames == kBlockSize` always holds true, why not just use `kBlockSize`? Is this some specific design choice?